File: CMOWebApi.Models/ComparetiveModel/EBookletModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CMOWebApi.Models.ComparetiveModel
{
	public class EBookletFilterModel
	{
		public int DepartmentCode { get; set; }
		public int BeneficiaryCategoryCode { get; set; }
		public int KPICategoryCode { get; set; }
        public long YearWiseCode { get; set; }
        public long GrandTotalCode { get; set; }
        public int GeneralEntryEBookletCode { get; set; }
		public int GeneralEntryEBookletVariableCode { get; set; }
        public long ParameterCategoryCode { get; set; }
    }

	public class EBookletResponseModel {
		public string Logo { get; set; }
		public string Name { get; set; }

		public List<BookletDataModel> EBookletList { get; set; }
	}

	public class BookletDataModel
	{
		public string Description { get; set; }
		public string YearName { get; set; }
	}

    #region Front

    public class EBookletCategoryModel
    {
        public int Code { get; set; }
        public string EnglishName { get; set; }
        public string HindiName { get; set; }
        public string LogoURL { get; set; }
    }

    public class EBookletCategoryFilterModel
    {
        public bool IsLogoURLBase64 { get; set; }
        public int Type { get; set; }
        public int GeneralEntryEBookletCode { get; set; }
        public int GeneralEntryEBookletVariableCode { get; set; }
    }

    public class FrontBookletDetailListModel
    {
        public string Description { get; set; }
        public string YearName { get; set; }
        public int TotalRecords { get; set; }
    }
    public class FrontBookletDetailModel
    {
        public Nullable<int> Code { get; set; }
        public string EnglishName { get; set; }
        public string HindiName { get; set; }
        public string LogoURL { get; set; }
        public List<FrontBookletDetailListModel> DetailListModel { get; set; }
    }

    public class EBookletFronFilterModel
    {
        public int DepartmentCode { get; set; }
        public int BeneficiaryCategoryCode { get; set; }
        public int KPICategoryCode { get; set; }
        public long YearWiseCode { get; set; }
        public long GrandTotalCode { get; set; }
        public int GeneralEntryEBookletCode { get; set; }
        public int GeneralEntryEBookletVariableCode { get; set; }
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public bool IsLogoURLBase64 { get; set; }
        public long ParameterCategoryCode { get; set; }
    }

    #endregion
}

File: ANGULAR/ANGULAR/src/app/Shared/Model/Camparetive/e-booklet-model.ts
import { environment } from 'src/environments/environment';
import { EbookletFilterProdEnum, EbookletFilterEnum, EbookletEnum, EbookletEnumProd } from '../../Enum/ebooklet.enum';

export class EBookletFilterModel {
  CategoryWiseRadio:number;
  DepartmentCode: number ;
  BeneficiaryCategoryCode: number ;
  KPICategoryCode: number;
  YearWiseCode:number;
  GrandTotalCode: number;
  GeneralEntryEBookletCode: number;
  GeneralEntryEBookletVariableCode: number;
  ParameterCategoryCode: number;
  constructor(){
   const ebookletEnum = environment.production?EbookletFilterProdEnum:EbookletFilterEnum;
   this.YearWiseCode=ebookletEnum.YearWiseCode;
   this.GrandTotalCode=ebookletEnum.GrandTotalCode;
   this.GeneralEntryEBookletCode=ebookletEnum.GeneralEntryEBookletCode;
   this.GeneralEntryEBookletVariableCode =ebookletEnum.GeneralEntryEBookletVariableCode;

   const enums =environment.production?EbookletEnumProd: EbookletEnum;
   this.ParameterCategoryCode=enums.EbookletCategory;
  }
}

export class EBookletResponseModel {
  Logo: string;
  Name: string;
  EBookletList: BookletDataModel[] = [];
}

export class BookletDataModel {
  Description: string;
  YearName: string;
}
